app/controllers/tags_controller.rb

```diff
@@ -90,23 +88,34 @@ def show
   end
 
   def feed
-    begin
-      @tag = Tag.find(params[:id])
-    rescue ActiveRecord::RecordNotFound
-       raise ActiveRecord::RecordNotFound, "Couldn't find tag with id '#{params[:id]}'"
-    end
-    if !@tag.canonical? && @tag.merger
-      @tag = @tag.merger
-    end
-    # Temp for testing
-    if %w(Fandom Character Relationship).include?(@tag.type.to_s) || @tag.name == "F/F"
-      if @tag.canonical?
-        @works = @tag.filtered_works.visible_to_all.order("created_at DESC").limit(25)
+    hash = Rails.cache.fetch(Tag.tag_feeds_key(params[:id])) do
+      path = nil
+      works = nil 
+      begin
+        tag = Tag.find(params[:id])
+      rescue ActiveRecord::RecordNotFound
+        raise ActiveRecord::RecordNotFound, "Couldn't find tag with id '#{params[:id]}'"
+      end
+      if !tag.canonical? && tag.merger
+        tag = tag.merger
+      end
+      # Temp for testing
+      if %w(Fandom Character Relationship).include?(tag.type.to_s) || tag.name == "F/F"
+        if tag.canonical?
+          works = tag.filtered_works.visible_to_all.order("created_at DESC").limit(25).all
+        else
+          works = tag.works.visible_to_all.order("created_at DESC").limit(25).all
+        end
       else
-        @works = @tag.works.visible_to_all.order("created_at DESC").limit(25)
+        path = tag_works_path(tag_id: tag.to_param)
       end
-    else
-      redirect_to tag_works_path(:tag_id => @tag.to_param) and return
+      { path: path, works: works, tag: tag }
+    end
+    redirect = hash[:path]
+    @works = hash[:works]
+    @tag = hash[:tag]
+    unless redirect.nil? 
+      redirect_to path and return
     end
```

Comment:
Doesn't this need to be redirect_to hash[:path]? I'm not sure how this is working: is there a test that is actually exercising this code? If not I suggest writing the test first, because I am pretty sure it should fail, and then fixing the code after. 

Also, why are you using this separate variable redirect? Why not just unless hash[:path].nil? And setting the @works and @tag is pointless if there is a redirect, isn't it? I would suggest moving that to after the unless check.